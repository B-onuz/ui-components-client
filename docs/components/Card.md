# Card (Component)

Componente de cartão customizável para dados

## Como utilizar

```js
import Card from 'ui-components/lib/components/Card'
```

```jsx
<Card title={'Demandas Pendentes'} count={99} color={'yellow'} icon={faArchive} />
<Card title={'Demandas Em Aberto'} count={99} color={'teal'} icon={faArchive} />
<Card title={'Demandas Concluídas'} count={99} color={'blue'} icon={faArchive} />
<Card title={'Demandas Canceladas'} count={99} color={'red'} icon={faArchive} />
```

## Variações

### Variant

```jsx
<Card variant={'variant'} />
```

Escala no tema: `theme.variantScale`

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
