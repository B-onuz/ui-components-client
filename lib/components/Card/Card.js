"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactFontawesome = require("@fortawesome/react-fontawesome");

var _styledSystem = require("styled-system");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Card/Card.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject7() {
  var data = _taggedTemplateLiteralLoose(["\n  width: 100%;\n  display: flex;\n  flex-direction: ", ";\n  justify-content: center;\n  align-items: center;\n  background-color: ", ";\n  border: none;\n  outline: none;\n  &:hover {\n    cursor: pointer;\n  }\n  &:focus {\n    box-shadow: 0px 0px 12px 4px ", ";\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-top: 8px;\n  display: ", ";\n  color: ", ";\n  @media only screen and (max-width: 500px) {\n    margin-top: 8px;\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  max-width: 50px;\n  padding: 4px;\n  background-color: ", ";\n  margin-right: ", ";\n  @media only screen and (max-width: 500px) {\n    margin-right: 0;\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  margin: 0;\n  color: ", ";\n  font-weight: 600;\n  box-shadow: none;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  justify-content: flex-start;\n  align-items: center;\n  color: ", ";\n  box-shadow: none;\n  font-size: 40px;\n  @media only screen and (max-width: 500px) {\n    font-size: 30px;\n    flex-direction: column;\n    align-items: center;\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  margin: 0;\n  text-transform: uppercase;\n  font-size: 13px;\n  font-weight: 400;\n  color: ", ";\n  opacity: ", ";\n  box-shadow: none;\n  margin-bottom: 8px;\n  @media only screen and (max-width: 500px) {\n    font-size: 13px;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  color: ", ";\n  width: 192px;\n  height: ", ";\n  min-height: ", ";\n  display: flex;\n  flex-direction: ", ";\n  justify-content: space-around;\n  position: relative;\n  box-shadow: 3px 3px 6px #00000029;\n  border: 3px solid;\n  border-radius: 12px;\n  padding: ", ";\n  overflow: ", ";\n  @media only screen and (max-width: 500px) {\n    width: 100%;\n    height: none !important;\n    min-height: ", ";\n    padding: ", ";\n    justify-content: center;\n    text-align: center;\n  }\n\n  .icon {\n    margin-right: ", ";\n    font-size: 40px;\n    @media only screen and (max-width: 500px) {\n      font-size: 30px;\n      margin: 0;\n      margin-bottom: 6px;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledCard = _styledComponents["default"].div(_templateObject(), function (_ref) {
  var theme = _ref.theme,
      color = _ref.color;
  return theme.colors[color] ? theme.colors[color] : color;
}, function (_ref2) {
  var clickableCard = _ref2.clickableCard;
  return !!clickableCard ? 'none' : '90px';
}, function (_ref3) {
  var clickableCard = _ref3.clickableCard,
      parentHeight = _ref3.parentHeight;
  return !!clickableCard ? parentHeight || '90px' : 'none';
}, function (_ref4) {
  var clickableCard = _ref4.clickableCard;
  return !!clickableCard ? 'row' : 'column';
}, function (_ref5) {
  var clickableCard = _ref5.clickableCard;
  return !!clickableCard ? 0 : '20px 32px';
}, function (_ref6) {
  var clickableCard = _ref6.clickableCard;
  return !!clickableCard ? 'visible' : 'hidden';
}, function (_ref7) {
  var parentHeight = _ref7.parentHeight;
  return !!parentHeight ? parentHeight : '84px !important';
}, function (_ref8) {
  var clickableCard = _ref8.clickableCard;
  return !!clickableCard ? 0 : '25px 12px';
}, function (_ref9) {
  var clickableCard = _ref9.clickableCard;
  return !!clickableCard ? 0 : '24px';
});

var StyledTitle = _styledComponents["default"].span(_templateObject2(), function (_ref10) {
  var theme = _ref10.theme,
      textColor = _ref10.textColor;
  return theme.colors[textColor] ? theme.colors[textColor] : textColor;
}, function (_ref11) {
  var cancel = _ref11.cancel;
  return cancel ? '60%' : 1;
});

var StyledInfo = _styledComponents["default"].div(_templateObject3(), function (_ref12) {
  var theme = _ref12.theme,
      iconColor = _ref12.iconColor;
  return theme.colors[iconColor] ? theme.colors[iconColor] : iconColor;
});

var Counter = _styledComponents["default"].span(_templateObject4(), function (_ref13) {
  var theme = _ref13.theme,
      counterColor = _ref13.counterColor;
  return theme.colors[counterColor] ? theme.colors[counterColor] : counterColor;
});

var StyledIcon = _styledComponents["default"].img(_templateObject5(), function (_ref14) {
  var theme = _ref14.theme,
      imgBackground = _ref14.imgBackground;
  return theme.colors[imgBackground] ? theme.colors[imgBackground] : imgBackground;
}, function (_ref15) {
  var clickableCard = _ref15.clickableCard;
  return !!clickableCard ? 0 : '24px';
});

var StyledHelper = _styledComponents["default"].small(_templateObject6(), function (_ref16) {
  var disclaimer = _ref16.disclaimer;
  return !!disclaimer ? 'inline-block' : 'none';
}, function (_ref17) {
  var theme = _ref17.theme,
      disclaimerColor = _ref17.disclaimerColor;
  return theme.colors[disclaimerColor] ? theme.colors[disclaimerColor] : disclaimerColor;
});

var ClickableCard = _styledComponents["default"].button(_templateObject7(), function (_ref18) {
  var reverseOrder = _ref18.reverseOrder;
  return !!reverseOrder ? 'column-reverse' : 'column';
}, function (_ref19) {
  var theme = _ref19.theme,
      cardBackground = _ref19.cardBackground;
  return theme.colors[cardBackground] ? theme.colors[cardBackground] : cardBackground;
}, function (_ref20) {
  var theme = _ref20.theme,
      cardBackground = _ref20.cardBackground;
  return theme.colors[cardBackground];
});

var Card = function Card(_ref21) {
  var title = _ref21.title,
      count = _ref21.count,
      icon = _ref21.icon,
      customIcon = _ref21.customIcon,
      customIconAlt = _ref21.customIconAlt,
      imgBackground = _ref21.imgBackground,
      cancel = _ref21.cancel,
      counterColor = _ref21.counterColor,
      iconColor = _ref21.iconColor,
      textColor = _ref21.textColor,
      disclaimer = _ref21.disclaimer,
      disclaimerColor = _ref21.disclaimerColor,
      clickableCard = _ref21.clickableCard,
      cardBackground = _ref21.cardBackground,
      reverseOrder = _ref21.reverseOrder,
      parentHeight = _ref21.parentHeight,
      rest = _objectWithoutPropertiesLoose(_ref21, ["title", "count", "icon", "customIcon", "customIconAlt", "imgBackground", "cancel", "counterColor", "iconColor", "textColor", "disclaimer", "disclaimerColor", "clickableCard", "cardBackground", "reverseOrder", "parentHeight"]);

  return /*#__PURE__*/_react["default"].createElement(StyledCard, _extends({}, rest, {
    clickableCard: clickableCard,
    parentHeight: parentHeight,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 131,
      columnNumber: 5
    }
  }), !!clickableCard ? /*#__PURE__*/_react["default"].createElement(ClickableCard, {
    cardBackground: cardBackground,
    reverseOrder: reverseOrder,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 133,
      columnNumber: 9
    }
  }, /*#__PURE__*/_react["default"].createElement(StyledTitle, {
    textColor: textColor,
    cancel: cancel,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 134,
      columnNumber: 11
    }
  }, title), icon && /*#__PURE__*/_react["default"].createElement(_reactFontawesome.FontAwesomeIcon, {
    icon: icon,
    className: "icon",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 137,
      columnNumber: 20
    }
  }), customIcon && /*#__PURE__*/_react["default"].createElement(StyledIcon, {
    src: customIcon,
    alt: customIconAlt,
    imgBackground: imgBackground,
    clickableCard: clickableCard,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 138,
      columnNumber: 26
    }
  })) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(StyledTitle, {
    textColor: textColor,
    cancel: cancel,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 142,
      columnNumber: 11
    }
  }, title), /*#__PURE__*/_react["default"].createElement(StyledInfo, {
    iconColor: iconColor,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 145,
      columnNumber: 11
    }
  }, icon && /*#__PURE__*/_react["default"].createElement(_reactFontawesome.FontAwesomeIcon, {
    className: "icon",
    icon: icon,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 146,
      columnNumber: 22
    }
  }), customIcon && /*#__PURE__*/_react["default"].createElement(StyledIcon, {
    src: customIcon,
    alt: customIconAlt,
    imgBackground: imgBackground,
    clickableCard: clickableCard,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 147,
      columnNumber: 28
    }
  }), /*#__PURE__*/_react["default"].createElement(Counter, {
    counterColor: counterColor,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 148,
      columnNumber: 13
    }
  }, count)), /*#__PURE__*/_react["default"].createElement(StyledHelper, {
    disclaimerColor: disclaimerColor,
    disclaimer: disclaimer,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 150,
      columnNumber: 11
    }
  }, disclaimer)));
};

Card.defaultProps = {
  textColor: 'black',
  disclaimerColor: 'black',
  clickableCard: false,
  reverseOrder: false
};
var _default = Card;
exports["default"] = _default;
module.exports = exports.default;