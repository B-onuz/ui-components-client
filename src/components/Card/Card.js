import React from 'react'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { background } from 'styled-system'

const StyledCard = styled.div`
  color: ${({ theme, color }) => (theme.colors[color] ? theme.colors[color] : color)};
  width: 192px;
  height: ${({ clickableCard }) => (!!clickableCard ? 'none' : '90px')};
  min-height: ${({ clickableCard, parentHeight }) => (!!clickableCard ? parentHeight || '90px' : 'none')};
  display: flex;
  flex-direction: ${({ clickableCard }) => (!!clickableCard ? 'row' : 'column')};
  justify-content: space-around;
  position: relative;
  box-shadow: 3px 3px 6px #00000029;
  border: 3px solid;
  border-radius: 12px;
  padding: ${({ clickableCard }) => (!!clickableCard ? 0 : '20px 32px')};
  overflow: ${({ clickableCard }) => (!!clickableCard ? 'visible' : 'hidden')};
  @media only screen and (max-width: 500px) {
    width: 100%;
    height: none !important;
    min-height: ${({ parentHeight }) => (!!parentHeight ? parentHeight : '84px !important')};
    padding: ${({ clickableCard }) => (!!clickableCard ? 0 : '25px 12px')};
    justify-content: center;
    text-align: center;
  }

  .icon {
    margin-right: ${({ clickableCard }) => (!!clickableCard ? 0 : '24px')};
    font-size: 40px;
    @media only screen and (max-width: 500px) {
      font-size: 30px;
      margin: 0;
      margin-bottom: 6px;
    }
  }
`

const StyledTitle = styled.span`
  margin: 0;
  text-transform: uppercase;
  font-size: 13px;
  font-weight: 400;
  color: ${({ theme, textColor }) => (theme.colors[textColor] ? theme.colors[textColor] : textColor)};
  opacity: ${({ cancel }) => (cancel ? '60%' : 1)};
  box-shadow: none;
  margin-bottom: 8px;
  @media only screen and (max-width: 500px) {
    font-size: 13px;
  }
`

const StyledInfo = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  color: ${({ theme, iconColor }) => (theme.colors[iconColor] ? theme.colors[iconColor] : iconColor)};
  box-shadow: none;
  font-size: 40px;
  @media only screen and (max-width: 500px) {
    font-size: 30px;
    flex-direction: column;
    align-items: center;
  }
`

const Counter = styled.span`
  margin: 0;
  color: ${({ theme, counterColor }) => (theme.colors[counterColor] ? theme.colors[counterColor] : counterColor)};
  font-weight: 600;
  box-shadow: none;
`

const StyledIcon = styled.img`
  max-width: 50px;
  padding: 4px;
  background-color: ${({ theme, imgBackground }) => (theme.colors[imgBackground] ? theme.colors[imgBackground] : imgBackground)};
  margin-right: ${({ clickableCard }) => (!!clickableCard ? 0 : '24px')};
  @media only screen and (max-width: 500px) {
    margin-right: 0;
  }
`

const StyledHelper = styled.small`
  margin-top: 8px;
  display: ${({ disclaimer }) => (!!disclaimer ? 'inline-block' : 'none')};
  color: ${({ theme, disclaimerColor }) => (theme.colors[disclaimerColor] ? theme.colors[disclaimerColor] : disclaimerColor)};
  @media only screen and (max-width: 500px) {
    margin-top: 8px;
  }
`

const ClickableCard = styled.button`
  width: 100%;
  display: flex;
  flex-direction: ${({ reverseOrder }) => (!!reverseOrder ? 'column-reverse' : 'column')};
  justify-content: center;
  align-items: center;
  background-color: ${({ theme, cardBackground }) => (theme.colors[cardBackground] ? theme.colors[cardBackground] : cardBackground)};
  border: none;
  outline: none;
  &:hover {
    cursor: pointer;
  }
  &:focus {
    box-shadow: 0px 0px 12px 4px ${({ theme, cardBackground }) => theme.colors[cardBackground]};
  }
`

const Card = ({
  title,
  count,
  icon,
  customIcon,
  customIconAlt,
  imgBackground,
  cancel,
  counterColor,
  iconColor,
  textColor,
  disclaimer,
  disclaimerColor,
  clickableCard,
  cardBackground,
  reverseOrder,
  parentHeight,
  ...rest
}) => {
  return (
    <StyledCard {...rest} clickableCard={clickableCard} parentHeight={parentHeight}>
      {!!clickableCard ? (
        <ClickableCard cardBackground={cardBackground} reverseOrder={reverseOrder}>
          <StyledTitle textColor={textColor} cancel={cancel}>
            {title}
          </StyledTitle>
          {icon && <FontAwesomeIcon icon={icon} className="icon" />}
          {customIcon && <StyledIcon src={customIcon} alt={customIconAlt} imgBackground={imgBackground} clickableCard={clickableCard} />}
        </ClickableCard>
      ) : (
        <>
          <StyledTitle textColor={textColor} cancel={cancel}>
            {title}
          </StyledTitle>
          <StyledInfo iconColor={iconColor}>
            {icon && <FontAwesomeIcon className="icon" icon={icon} />}
            {customIcon && <StyledIcon src={customIcon} alt={customIconAlt} imgBackground={imgBackground} clickableCard={clickableCard} />}
            <Counter counterColor={counterColor}>{count}</Counter>
          </StyledInfo>
          <StyledHelper disclaimerColor={disclaimerColor} disclaimer={disclaimer}>
            {disclaimer}
          </StyledHelper>
        </>
      )}
    </StyledCard>
  )
}

Card.defaultProps = {
  textColor: 'black',
  disclaimerColor: 'black',
  clickableCard: false,
  reverseOrder: false,
}

export default Card
