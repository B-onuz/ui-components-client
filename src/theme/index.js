const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
