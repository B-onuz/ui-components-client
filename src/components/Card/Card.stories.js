import React from 'react'
import Card from './Card'
import { Flex, Box } from 'reflexbox'
import { faArchive, faBoxes, faBoxOpen, faParachuteBox, faBan, faThumbsUp } from '@fortawesome/free-solid-svg-icons'
import serv_waiting from '../../assets/serv_waiting.svg'
import Typography from '../Typography'

export default {
  title: 'Card',
  component: Card,
}

export const CardExample = () => (
  <>
    <Card title={'Demandas Pendentes'} count={99} color={'danger'} icon={faArchive} />
    <Card
      title={
        <Typography textAlign="left" m={0}>
          REVISÃO DE OUTROS DOCUMENTOS OU CONTRATOS
        </Typography>
      }
      color="purple"
      icon={faParachuteBox}
      disclaimer={
        <Typography m={0} textAlign="left">
          Pendente
        </Typography>
      }
      disclaimerColor="lightGrey"
      count={
        <Typography m={0} fontSize={2}>
          {`#58758`}
        </Typography>
      }
    />
  </>
)

export const CardColors = () => (
  <Flex justifyContent={'flex-start'} p={3} flexWrap={'wrap'}>
    <Card title={'Demandas Pendentes'} count={99} color={'lightGrey'} icon={faBoxes} />
    <Card title={'Demandas Em Aberto'} count={99} color={'teal'} icon={faBoxOpen} />
    <Card title={'Demandas Concluídas'} count={13279} color={'green'} icon={faThumbsUp} />
    <Card title={'Demandas Canceladas'} count={99} cancel={true} color={'red'} icon={faBan} />
    <Card title={'Pacotes promocionais'} count={99} color={'blue'} icon={faParachuteBox} disclaimer={'últimos 7 dias'} disclaimerColor={'#138204'} />
  </Flex>
)

export const MultiColorsControl = () => (
  <Flex>
    <Card
      color={'primary'}
      textColor={'#DDD'}
      iconColor={'#000'}
      counterColor={'warning'}
      title={'Demandas Pendentes'}
      count={99}
      customIcon={serv_waiting}
      // imgBackground={'primary'}
    />
    <Card color={'primary'} textColor={'#DDD'} iconColor={'#DDD'} title={'Demandas Pendentes'} icon={faParachuteBox} cardBackground={'primary'} />
  </Flex>
)

export const CardButton = () => (
  <>
    <Flex>
      <Card
        clickableCard
        color={'primary'}
        textColor={'#DDD'}
        iconColor={'#DDD'}
        title={'Demandas Pendentes'}
        customIcon={serv_waiting}
        cardBackground={'primary'}
        mr={2}
      />
      <Card
        clickableCard
        reverseOrder
        color={'primary'}
        textColor={'#DDD'}
        iconColor={'#DDD'}
        title={'Demandas Pendentes'}
        customIcon={serv_waiting}
        cardBackground={'primary'}
        parentHeight="115px"
      />
    </Flex>
    <Box mt={[3]}>
      <Card
        clickableCard
        reverseOrder
        color={'primary'}
        textColor={'#DDD'}
        iconColor={'#DDD'}
        title={'Demandas Pendentes'}
        customIcon={serv_waiting}
        cardBackground={'primary'}
      />
    </Box>
  </>
)
