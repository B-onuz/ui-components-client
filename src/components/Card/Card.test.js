import React from 'react'
import Card from './Card.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render Card', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Card />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
