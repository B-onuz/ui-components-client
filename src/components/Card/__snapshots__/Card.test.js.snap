// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Card 1`] = `
.c0 {
  width: 192px;
  height: 90px;
  min-height: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  position: relative;
  box-shadow: 3px 3px 6px #00000029;
  border: 3px solid;
  border-radius: 12px;
  padding: 20px 32px;
  overflow: hidden;
}

.c0 .icon {
  margin-right: 24px;
  font-size: 40px;
}

.c1 {
  margin: 0;
  text-transform: uppercase;
  font-size: 13px;
  font-weight: 400;
  color: #1b1c1d;
  opacity: 1;
  box-shadow: none;
  margin-bottom: 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: none;
  font-size: 40px;
}

.c3 {
  margin: 0;
  font-weight: 600;
  box-shadow: none;
}

.c4 {
  margin-top: 8px;
  display: none;
  color: #1b1c1d;
}

@media only screen and (max-width:500px) {
  .c0 {
    width: 100%;
    height: none !important;
    min-height: 84px !important;
    padding: 25px 12px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    text-align: center;
  }
}

@media only screen and (max-width:500px) {
  .c0 .icon {
    font-size: 30px;
    margin: 0;
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:500px) {
  .c1 {
    font-size: 13px;
  }
}

@media only screen and (max-width:500px) {
  .c2 {
    font-size: 30px;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media only screen and (max-width:500px) {
  .c4 {
    margin-top: 8px;
  }
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <div
    className="c2"
  >
    <span
      className="c3"
    />
  </div>
  <small
    className="c4"
  />
</div>
`;
